extension with LoV (#731)

FredLL-Avaiga opened the pull request:
- RN: Support inner_properties for Element: properties that are only internal and used to handle communication between the backend and the frontend (no user visibility)
- refresh typed variable (_TaipyBase)
- LoV adapter: allow extra properties in LoV element

Commit:
#731 extension with LoV

FredLL-Avaiga committed:
Update gui.pyi file

Commit:
don't break the API

Commit:
Merge branch 'feature/#731-extension-with-LoV' of https://github.com/Avaiga/taipy-gui into feature/#731-extension-with-LoV

Commit:
format

dinhlongviolin1 reviewed (approved):
LGTM

Commit:
add TaipyBase extension support

FredLL-Avaiga committed:
Update gui.pyi file

Commit:
check unicity

Commit:
Merge branch 'feature/#731-extension-with-LoV' of https://github.com/Avaiga/taipy-gui into feature/#731-extension-with-LoV

Commit:
for pyi file ?

FredLL-Avaiga committed:
Update gui.pyi file

FabienLelaquais reviewed (approved):
I get it now.
Well done!

FredLL-Avaiga commented:
@dinhlongviolin1 Any idea why gui.pyi file is causing pb with mypy ?

dinhlongviolin1 committed:
Empty commit to trigger actions

dinhlongviolin1 committed:
Merge branch 'develop' into feature/#731-extension-with-LoV

Commit:
Handle Deprecation Warning

Commit:
Merge branch 'feature/#731-extension-with-LoV' of https://github.com/Avaiga/taipy-gui into feature/#731-extension-with-LoV

Commit:
Handle Deprecation Warning in debug

Commit:
don't warn when value is None

Commit:
define and use a specific warning for Taipy Gui

Commit:
report the right source

Commit:
fix tests for warnings

Commit:
add _warnings to conftest

Commit:
sort

Commit:
- _warnings.warnings_warn renamed to _warnings._warn
- Some warning rephrased (+conventions applied).
- Hide the Config GUI class and is_debugging().

FredLL-Avaiga commented on src/taipy/gui/config.py:
```diff
@@ -159,9 +159,7 @@ def _get_config(self, name: ConfigParameter, default_value: t.Any) -> t.Any:  #
-                    warnings_warn(
-                        f'app_config "{name}" value "{self.config[name]}" is not of type {type(default_value)}\n{e}'
-                    )
+                    _warn(f'app_config "{name}" value "{self.config[name]}" is not of type {type(default_value)}\n{e}')
```
no :\n{e} ?

FabienLelaquais replied:
How and why did I miss that?!?
Thx!

FredLL-Avaiga commented on src/taipy/gui/utils/date.py:
```diff
@@ -26,7 +26,7 @@ def _date_to_ISO(date_val: t.Union[datetime, date, time]) -> str:
             # astimezone() fails on Windows for pre-epoch times
             # See https://bugs.python.org/issue36759
-            warnings_warn(f"There is some problems with date parsing to ISO!\n{e}")
+            _warn(f"There is some problems with date parsing to ISO!\n{e}")
```
! => :

FabienLelaquais replied:
👌

FredLL-Avaiga commented on src/taipy/gui/config.py:
```diff
@@ -227,7 +225,7 @@ def _build_config(self, root_dir, env_filename, kwargs):  # pragma: no cover
                 except Exception as e:
-                    warnings_warn(
+                    _warn(
                         f"Invalid keyword arguments value in Gui.run {key} - {value}. Unable to parse value to the correct type.\n{e}"
```
:

FabienLelaquais replied:
👌

FredLL-Avaiga commented on src/taipy/gui/config.py:
```diff
@@ -238,8 +236,8 @@ def _build_config(self, root_dir, env_filename, kwargs):  # pragma: no cover
-                        warnings_warn(
-                            f"Invalid env value in Gui.run: {key} - {value}. Unable to parse value to the correct type.\n{e}"
+                        _warn(
+                            f"Invalid env value in Gui.run(): {key} - {value}. Unable to parse value to the correct type.\n{e}"
```
:

Commit:
- Fix test

Commit:
Comments from FredLL
